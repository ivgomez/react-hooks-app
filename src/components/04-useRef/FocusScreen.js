import React, { useRef } from "react";
import "../02-useEffect/effects.css";

export const FocusScreen = () => {
  const inputRef = useRef();
  const handleClick = () => {
    inputRef.current.select();
  };

  return (
    <div>
      <h1>Focus</h1>
      <hr />
      <div className="form-group">
        <input ref={inputRef} className="form-control" placeholder="Name" />
      </div>
      <div />
      <button className="btn btn-primary" onClick={handleClick}>
        Save
      </button>
    </div>
  );
};
