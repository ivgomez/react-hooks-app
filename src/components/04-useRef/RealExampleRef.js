import React, { useState } from "react";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";
import "../02-useEffect/effects.css";

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);
  return (
    <div>
      <h1>Real Example Ref</h1>
      <hr />
      {show && (
        <div className="form-group">
          <MultipleCustomHooks />
        </div>
      )}
      <button className="btn btn-primary" onClick={() => setShow(!show)}>
        Show/Hide
      </button>
    </div>
  );
};
