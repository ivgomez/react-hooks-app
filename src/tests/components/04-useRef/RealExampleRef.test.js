import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe("test on <RealExample>", () => {
  const wrapper = shallow(<RealExampleRef />);
  test("it should rendered ok", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });
  test("it should rendered the component <MultipleCustomHooks />", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
