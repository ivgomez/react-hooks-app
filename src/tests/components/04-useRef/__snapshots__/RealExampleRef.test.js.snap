// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <RealExample> it should rendered ok 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
